### ISSUE TITLE
Rule Change: `require-v-flag` boolean option in `require-unicode-regexp`

### ISSUE THREAD

**Issue by brettz9** [eslint/eslint#18712]
### What rule do you want to change?

require-unicode-regexp

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
/abc/u
new RegExp('abc', 'u')
```

### What does the rule currently do for this code?

Passes it

### What will the rule do after it's changed?

Fails it

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

Although it is early, for those whose targeted browser/server versions allow, one can better ensure one is using consistent syntax now (and is not later forced to update syntax which fails with the `v` flag but not the `u` flag).

**Comment by nzakas** [comment]
`require-unicode-regexp` already allows the `v` flag to prevent a warning, so I'm not sure what it that you're asking for. Can you explain with more than a sentence?

**Comment by brettz9** [comment]
I would like it for *only* the `v` flag to prevent the warning—not the `u` flag.

**Comment by nzakas** [comment]
Okay, so you want to enforce that only `v` can satisfy this rule. Can you explain why someone might want that?

And is there a case where someone might want only `u` to satisfy the rule?

**Comment by nzakas** [comment]
Thanks, that's helpful background. I think adding an option makes sense, but we probably do need to make it go both ways as people might want to enforce `u` if they're in older ES environments. So maybe the option would be `requireFlag` with possible values of `"u"` and `"v"`.

@eslint/eslint-team would appreciate some other thoughts on this.

**Comment by Tanujkanti4441** [comment]
i am with the approach of adding `requiedFlag` option for enforcing one of the flag for consistency and as this rule gives fix by adding `u` flag we have to update that also according to the option.

and i also noticed that rule doesn't report if we have both `u` and `v` flag together as following
> const c = new RegExp("ccc", "uv");

new option can help report this also.

**Comment by mdjermanovic** [comment]
Adding an option to enforce the use of the `v` flag sounds good to me.

I'm not sure if a core rule should have an option to enforce the `u` flag over the `v` flag. Disallowing new language features seems better suited for environment-specific plugins like mentioned `eslint-plugin-escompat` and `eslint-plugin-n`, and there's also a generic plugin [`eslint-plugin-es-x`](https://github.com/eslint-community/eslint-plugin-es-x) that already has rule [`es-x/no-regexp-v-flag`](https://eslint-community.github.io/eslint-plugin-es-x/rules/no-regexp-v-flag.html).

**Comment by mdjermanovic** [comment]
> and i also noticed that rule doesn't report if we have both `u` and `v` flag together as following
> 
> > const c = new RegExp("ccc", "uv");
> 
> new option can help report this also.

This is reported by the `no-invalid-regexp` rule: [Playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8taW52YWxpZC1yZWdleHA6IFwiZXJyb3JcIiovXG5cbmNvbnN0IGMgPSBuZXcgUmVnRXhwKFwiY2NjXCIsIFwidXZcIik7Iiwib3B0aW9ucyI6eyJydWxlcyI6e30sImxhbmd1YWdlT3B0aW9ucyI6eyJzb3VyY2VUeXBlIjoibW9kdWxlIiwicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==)

**Comment by nzakas** [comment]
> I'm not sure if a core rule should have an option to enforce the `u` flag over the `v` flag.

To the extent that someone needs to support an older ECMAScript version, and they're using the `RegExp` constructor rather than a literal (so no parsing error), it seems helpful to be able to say that only "u" should be used. Otherwise, the only other option is to disable the rule completely.

**Comment by brettz9** [comment]
I've submitted #18836 if that may help toward the issue...

**Comment by mdjermanovic** [comment]
> To the extent that someone needs to support an older ECMAScript version, and they're using the `RegExp` constructor rather than a literal (so no parsing error), it seems helpful to be able to say that only "u" should be used. Otherwise, the only other option is to disable the rule completely.

I still think that disallowing new syntax or new language features is more suitable for plugins mentioned in https://github.com/eslint/eslint/issues/18712#issuecomment-2258941022. But since we're already adding an option to require a specific flag, I'm not opposed to the solution described in https://github.com/eslint/eslint/issues/18712#issuecomment-2258472918 (`"requireFlag": "u"|"v"`), so marking as accepted.
